app/services/classification/base.rb

```diff
@@ -178,11 +179,17 @@ def update_generic_matcher(lookup_values, options)
       computed_lookup_value
     end
 
-    def update_array_matcher(should_avoid_duplicates, lookup_values, options)
+    def update_array_matcher(default, should_avoid_duplicates, lookup_values, options)
       elements = []
       values = []
       element_names = []
 
+      if default.present?
```

Comment:
I think `unless default.nil?` would make more sense, what if default value is empty string?